File: 탐색 & 시뮬레이션/봉우리.py
import sys
# sys.stdin = open("input.txt", "rt")

'''
n = int(input())
arr = []
zeros = [0] * (n + 2)
arr.append(zeros)
for _ in range(n):
    tmp = [0] + list(map(int, input().split())) + [0]
    arr.append(tmp)
arr.append(zeros)

cnt = 0
for i in range(1, n + 1):
    for j in range(1, n + 1):
        comp = [arr[i - 1][j],
                arr[i + 1][j],
                arr[i][j - 1],
                arr[i][j + 1]]
 
        for element in comp:
            if arr[i][j] <= element:
                break
        else:
            cnt += 1
print(cnt)
'''

n = int(input())
arr = [list(map(int, input().split())) for _ in range(n)]
arr.insert(0, [0] * n)
arr.append([0] * n)
for row in arr:
    row.insert(0, 0)
    row.append(0)

di = [-1, 0, 1, 0]
dj = [0, 1, 0, -1]

cnt = 0
for i in range(1, n + 1):
    for j in range(1, n + 1):
        if all(arr[i][j] > arr[i + di[k]][j + dj[k]] for k in range(4)):
            cnt += 1

print(cnt)            

File: 정렬 & 이분탐색(결정알고리즘) & 투포인트 알고리즘 & 스택/봉우리.cpp
#include <stdio.h>
#include <vector>
using namespace std;

int main(void){
	// freopen("input.txt", "r", stdin);
	int n, i, j, k, pivot, cnt;
	int dx[4] = {-1, 0, 1, 0};
	int dy[4] = {0, 1, 0, -1};
	bool isHighest;
	scanf("%d", &n);
	vector<vector<int>> map(n + 2, vector<int> (n + 2));
	
	for(i = 1; i <= n; i++){
		for(j = 1; j <= n; j++){
			scanf("%d", &map[i][j]);	
		}
	}
	cnt = 0;
	for(i = 1; i <= n; i++){
		for(j = 1; j <= n; j++){
			pivot = map[i][j];
			isHighest = true;
			for(k = 0; k < 4; k++){
				if(map[i + dx[k]][j + dy[k]] >= pivot){
					isHighest = false;
					break;
				}
			}
			if(isHighest) cnt++;
		}
	}
	printf("%d", cnt);
	
	return 0;
}
